### PR TITLE
sys/ztimer/xtimer_compat: fix bug introduced in #17690

### PR DESCRIPTION
**Pull request by kfessel**
### Contribution description

while working on #17693 a bug in #17690 was detected

`xtimer_usleep(x)` -which redirects to `ztimer_sleep(ZTIMER_USEC,x)` needs no `clock` in `xtimer_tsleep64()`

also adds  missing
```
    dummy XTIMER_BACKOFF
    xtimer_init
    xtimer_is_set
    xtimer_rmutex_lock_timeout
```
### Testing procedure

build xtimer_test using ztimer/xtimer_compat

either by include of ztimer/xtimer_compat.h and adding ztimer_usec to USEMODULE

or adding 

ztimer_xtimer_compat to USEMODULE

### Issues/PRs references

while working on #17693 a bug in #17690 was detected

### CODE REVIEW

File: sys/include/ztimer/xtimer_compat.h [hunk 1]
```diff
@@ -197,6 +216,15 @@ static inline int xtimer_mutex_lock_timeout(mutex_t *mutex, uint64_t us)
     return 0;
 }
 
+static inline int xtimer_rmutex_lock_timeout(rmutex_t *rmutex, uint64_t us) {
```
Comment (fjmolinas) [inline sys/include/ztimer/xtimer_compat.h:219]:
```suggestion
static inline int xtimer_rmutex_lock_timeout(rmutex_t *rmutex, uint64_t us)
{
```

**Review by fjmolinas — COMMENTED** [review]
YOu can squash right away

File: sys/include/ztimer/xtimer_compat.h (hunk 1 above)
Comment (kfessel) [reply sys/include/ztimer/xtimer_compat.h:219]: Murdock run just started -> lets wait

Comment (kfessel) [reply sys/include/ztimer/xtimer_compat.h:219]: image

**Comment by kfessel** [comment]
image

**Review by fjmolinas — APPROVED** [review]
ACK!

**Pull request merged**